### PR TITLE
LUI-133: Fix error on module load when module uses Spring-injected filters

### PR DESCRIPTION
**Pull request by kwburnett**
Ensuring the Spring context has been refreshed before filters get loaded in case filters need dependencies defined in the Spring context. Previously, filters would be loaded before the Spring context was refreshed. This would lead to issues when the filters had a class of org.springframework.web.filter.DelegatingFilterProxy, meaning Spring is supposed to resolve all dependencies of the filter. If the context had not been refreshed, no beans would be found.

Additionally, an issue was also occurring when no "dwr-invoker" servlet was present, so made the reInitServlet only called if a servlet was present.

See this discussion on OpenMRS Talk for more clarification.
https://talk.openmrs.org/t/loading-modules-error-due-to-filters-and-spring-beans/14162

Here is the link to the JIRA item.
https://issues.openmrs.org/browse/LUI-133

### CODE REVIEW

**Comment by dkayiwa** [comment]
@kwburnett can you create a ticket for this https://wiki.openmrs.org/display/docs/Tickets and then reference the ticket url here as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by dkayiwa** [comment]
@kwburnett did you just forget to include a link to the ticket?

**Comment by kwburnett** [comment]
@dkayiwa I had missed that part. Thanks for the correction.

**Comment by dkayiwa** [comment]
@kwburnett your link seems to list more than one ticket. :)

**Comment by kwburnett** [comment]
Ah, you're right. It should show just the one now. :-)

File: omod/src/main/java/org/openmrs/module/web/controller/ModuleListController.java [hunk 1]
```diff
@@ -184,18 +184,18 @@ protected ModelAndView onSubmit(HttpServletRequest request, HttpServletResponse
 								ModuleFactory.startModule(depMod);
 								boolean thisModuleCausesRefresh = WebModuleUtil.startModule(depMod, getServletContext(), true);
 								someModuleNeedsARefresh = someModuleNeedsARefresh || thisModuleCausesRefresh;
-								if (someModuleNeedsARefresh) {
```
Comment (dkayiwa) [inline omod/src/main/java/org/openmrs/module/web/controller/ModuleListController.java]: This section of loading servlets and filters was called for each depMod in the for loop. Isn't removing it the same as losing functionality? 😊

**Comment by dkayiwa** [comment]
Do you mind squashing these commits into one as advised at? https://wiki.openmrs.org/display/docs/Pull+Request+Tips

**Comment by kwburnett** [comment]
@dkayiwa I followed the instructions on https://wiki.openmrs.org/display/docs/Pull+Request+Tips. But, when I squished and then pushed, it just added a new commit to this PR (the most recent one). Do I need to reset HEAD and then force override when I push?

**Comment by dkayiwa** [comment]
Do not fear to try the proposals you have 😊

**Comment by kwburnett** [comment]
Sawa sawa. Fixed.

File: pom.xml [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 	<groupId>org.openmrs.module</groupId>
 	<artifactId>legacyui</artifactId>
-	<version>1.3.3</version>
```
Comment (dkayiwa) [inline pom.xml]: Since when did this change from https://github.com/openmrs/openmrs-module-legacyui/blob/master/pom.xml#L18

File: api/pom.xml [hunk 3]
```diff
@@ -15,7 +15,7 @@
 	<parent>
 		<groupId>org.openmrs.module</groupId>
 		<artifactId>legacyui</artifactId>
-		<version>1.3.3</version>
```
Comment (dkayiwa) [inline api/pom.xml]: Since when did this change from https://github.com/openmrs/openmrs-module-legacyui/blob/master/api/pom.xml#L18

File: omod/pom.xml [hunk 4]
```diff
@@ -15,7 +15,7 @@
 	<parent>
 		<groupId>org.openmrs.module</groupId>
 		<artifactId>legacyui</artifactId>
-		<version>1.3.3</version>
+		<version>1.4.0-SNAPSHOT</version>
```
Comment (dkayiwa) [inline omod/pom.xml]: Since when did this change from https://github.com/openmrs/openmrs-module-legacyui/blob/master/omod/pom.xml#L18

File: omod/src/main/webapp/resources/css/openmrs.css [hunk 5]
```diff
@@ -1,4 +1,6 @@
 html {
+	font-family: Verdana, 'Lucida Grande', 'Trebuchet MS', Arial, Sans-Serif;
+    -moz-box-sizing: border-box;  /* Use IE-like Border-Box model */
```
Comment (dkayiwa) [inline omod/src/main/webapp/resources/css/openmrs.css]: Were these changes intentional?

**Comment by dkayiwa** [comment]
I see that you have a couple more changes that needed for this ticket. If squashing ends up giving you a hard time, you may find it easier to just switch to the master branch, create a new branch and switch to it, finally manually apply the changes, since they are not many.

**Comment by kwburnett** [comment]
Sorry, @dkayiwa. I'm messing this up big time. Anyways, let me close this and create a new PR.
